geomstats/geometry/discrete_curves.py

```diff
@@ -572,6 +618,47 @@ def dist(self, point_a, point_b, **kwargs):
 
         return dist
 
+    # TODO (alebrigant): generalize from_vector_to_diagonal_matrix to k
+    # diagonals and modify tensorflow vstack and hstack.
+    def space_derivative(self, curve):
+        """Compute space derivative of curve using centered differences.
+
+        Parameters
+        ----------
+        curve : array-like, shape=[..., n_points, ambient_dim]
+            Discrete curve.
+
+        Returns
+        -------
+        space_deriv : array-like, shape=[...,n_points, ambient_dim]
+        """
+        n_points = curve.shape[-2]
+
+        vec_1 = gs.squeeze(gs.hstack((
+            gs.array([[-1.]]),
+            gs.zeros((1, n_points - 2)),
```

Comment:
Did you make sure somewhere that `n_points >=2`?

Reply:
Good point.